tools/management/commands/assign_structure_states.py

```diff
@@ -243,21 +236,97 @@ def handle(self, *args, **options):
 
                         # Definitely an inactive state structure When distance is smaller than 13Å
                         if entry[1] < 13*distance_scaling_factor:
-                            struct.state, created = ProteinState.objects.get_or_create(slug="inactive", defaults={'name': "Inactive"})
+                            struct.state = ProteinState.objects.get_or_create(slug="inactive", defaults={'name': "Inactive"})[0]
 
                         # UGLY: knowledge-based hardcoded corrections
                         if entry[0] in hardcoded:
                             structure_state = hardcoded[entry[0]]
-                            struct.state, created = ProteinState.objects.get_or_create(slug=structure_state, defaults={'name': structure_state.capitalize()})
+                            struct.state = ProteinState.objects.get_or_create(slug=structure_state, defaults={'name': structure_state.capitalize()})[0]
 
                         # Save changes
                         struct.save()
 
-        self.logger.info('DONE assiging the "Degree Active" levels and activation states')
+        self.logger.info("DONE assiging the \"Degree Active\" levels and activation states")
+
+        if not options["states_only"]:
+            # Assigning the representative tag for the "best" structure with the same receptor-state combinations
+            self.logger.info("ASSIGNING the \"representative\" tag for unique structure-state complexes")
+
+            # Set the representative state of all GPCR structure to False
+            Structure.objects.filter(protein_conformation__protein__family__slug__startswith="00").update(representative=False)
+
+
+            # Select all GPCR structures and get unique slug-state combinations
+            struct_combs = list(Structure.objects.filter(protein_conformation__protein__family__slug__startswith="00") \
+                                .values_list("protein_conformation__protein__family__slug", "state", "pk", "resolution", "protein_conformation__pk", "protein_conformation__protein__parent__pk"))
```

Comment:
Maybe go one level higher with protein_conformation__protein__parent__pk to protein_conformation__protein__parent__family__pk to check on ortholog level instead of single receptor?

Reply:
So just to double check, you are suggesting to do the wildtype GN selection at the family level instead of the receptor level?

Reply:
No, I might have left the comment in the wrong place. I'm just suggesting not to have an active representative opsd_bovin and a separate active representative opsd_human but have have one representative for both

Reply:
Ah ok, in that case that's also what I was thinking and implemented. The check for unique combinations is done using the "combo_id" which is the combination of protein "slug + state".